pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -1007,6 +1009,15 @@ func (d *awsInstance) Addresses() map[string]v1.NodeAddressType {
 		aws.StringValue(d.instance.PrivateDnsName):   v1.NodeInternalDNS,
 	}
 
+	for _, netInterface := range d.instance.NetworkInterfaces {
+		for _, addr := range netInterface.Ipv6Addresses {
+			ipAddr := aws.StringValue(addr.Ipv6Address)
```

Comment:
Wouldn't this work without the casting? 
```suggestion
			ipAddr := addr.String()
```